### PR TITLE
refactor(proteus): Don't generate empty keys in KeyPair constructor

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: packages/proteus/src/main/keys/KeyPair.ts [hunk 1]
```diff
@@ -79,26 +79,24 @@ export class KeyPair {
...
+      const secretKey = SecretKey.decode(decoder);
+
+      decoder.u8();
+      const publicKey = PublicKey.decode(decoder);
+
+      return new KeyPair(publicKey, secretKey);
     }
 
-    return self;
+    throw new Error(`Unexpected number of properties: "${properties}"`);
```
Comment (ffflorian) [inline packages/proteus/src/main/keys/KeyPair.ts:100]: Let's use a `DecodeError` here, shall we?

File: packages/proteus/src/main/session/SendChain.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@
  */
 
 import * as CBOR from '@wireapp/cbor';
-
+import {PublicKey, SecretKey} from '../keys';
```
Comment (ffflorian) [inline packages/proteus/src/main/session/SendChain.ts:21]:
```suggestion

import {PublicKey, SecretKey} from '../keys';
```

Comment (bennyn) [reply packages/proteus/src/main/session/SendChain.ts:21]: We can drop these extra lines.

File: packages/proteus/src/main/keys/KeyPair.ts (hunk 1 above)
Comment (bennyn) [reply packages/proteus/src/main/keys/KeyPair.ts]: Great idea.

**Pull request merged**